lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/context/ContextRepresenters.java

```diff
@@ -53,8 +56,22 @@
   @Inject
   private ContextRepresenters(final EvaluatorMessageDispatcher messageDispatcher,
                               final ContextFactory contextFactory) {
+    this(messageDispatcher, contextFactory, Optional.<DriverRestartManager>empty());
+  }
+
+  @Inject
+  private ContextRepresenters(final EvaluatorMessageDispatcher messageDispatcher,
+                              final ContextFactory contextFactory,
+                              final DriverRestartManager driverRestartManager) {
+    this(messageDispatcher, contextFactory, Optional.of(driverRestartManager));
+  }
+
+  private ContextRepresenters(final EvaluatorMessageDispatcher messageDispatcher,
+                              final ContextFactory contextFactory,
+                              final Optional<DriverRestartManager> driverRestartManager) {
```

Comment:
This seems quite complicated. Have you considered a default implementation of `DriverRestartManager` that implements all its methods for the case that the Driver was *not*v restarted? That way, you can assume its presence in all other classes.